nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java

```diff
@@ -607,6 +612,25 @@ private WebAppContext loadWar(final File warFile, final String contextPath, fina
         return webappContext;
     }
 
+    private void addContentLengthFilter(String contextPath, String pathSpec, WebAppContext webappContext) {
+        final FilterHolder holder = new FilterHolder(ContentLengthFilter.class);
+        final Map<String, String> largePaths = props.getWebMaxContentSizeLargePaths();
+        int size;
+
+        if (!largePaths.containsValue(contextPath)) {
+            size = DataUnit.parseDataSize(props.getWebMaxContentSize(), DataUnit.B).intValue();
+        } else {
+            size = DataUnit.parseDataSize(props.getWebMaxContentSizeLarge(), DataUnit.B).intValue();
+        }
+        holder.setInitParameters(new HashMap<String, String>() {{
+            put("maxContentLength", String.valueOf(size));
+        }});
+
+        logger.info("Adding Content Length Filter to context at pathSpec: " + contextPath + " with max size: " + size + "b");
+        // NB: the use of pathSpec vs contextPath
```

Comment:
What is the difference between `pathSpec` and `contextPath` here?

Reply:
simplified and removed the extra argument.